File: src/main/java/com/oshi/libgenericsettings/viewholder/SettingsTitleSecondaryTitleViewHolder.java
package com.oshi.libgenericsettings.viewholder;

import android.databinding.ViewDataBinding;

import com.oshi.libgenericsettings.BR;
import com.oshi.libgenericsettings.data.TitleSecondaryTitleData;
import com.oshi.libgenericsettings.presenter.ISettingsPresenter;


public class SettingsTitleSecondaryTitleViewHolder extends BaseSettingsViewHolder<TitleSecondaryTitleData> {

    public SettingsTitleSecondaryTitleViewHolder(ViewDataBinding viewDataBinding) {
        super(viewDataBinding);
    }

    @Override
    public void onBind(TitleSecondaryTitleData data, ISettingsPresenter presenter, int position) {
        viewDataBinding.setVariable(BR.titleSecondaryTitleData, data);
        viewDataBinding.setVariable(BR.itemPosition, position);
        viewDataBinding.setVariable(BR.settingsPresenter, presenter);
        viewDataBinding.executePendingBindings();
    }
}

File: src/main/java/com/oshi/libgenericsettings/viewholder/SettingsTitleSecondaryTitleViewHolder.kt
package com.oshi.libgenericsettings.viewholder

import android.databinding.ViewDataBinding
import com.oshi.libgenericsettings.BR
import com.oshi.libgenericsettings.data.TitleSecondaryTitleData
import com.oshi.libgenericsettings.presenter.ISettingsPresenter

class SettingsTitleSecondaryTitleViewHolder(viewDataBinding: ViewDataBinding) : BaseSettingsViewHolder<TitleSecondaryTitleData>(viewDataBinding) {

    override fun onBind(data: TitleSecondaryTitleData, presenter: ISettingsPresenter, position: Int) {
        viewDataBinding.setVariable(BR.titleSecondaryTitleData, data)
        viewDataBinding.setVariable(BR.itemPosition, position)
        viewDataBinding.setVariable(BR.settingsPresenter, presenter)
        viewDataBinding.executePendingBindings()
    }

}